n8n playbook (#45)
amd/playbooks: playbooks/core/n8n-automation-gpt-oss/README.md

## Overview

n8n is a workflow automation platform that lets you connect apps and services using a visual node-based editor. Instead of writing code, you build workflows by dragging, connecting, and configuring nodes.

> This playbook teaches you how to set up an AI-powered financial news summarizer that scrapes the Financial Times website, extracts key headlines, and uses a local LLM running on your STX Halo™ to generate an investor-focused summary.

Reviewer: ```suggestion
This playbook teaches you how to set up an AI-powered financial news summarizer that scrapes the Financial Times website, extracts key headlines, and uses a local LLM running on your Ryzen™ AI Halo to generate an investor-focused summary.
```
Halo isn't trademarks, put in a suggestion based on how they're referring to the product. 
If you prefer STX Halo, just remove the trademark.

## What You'll Learn

- How to install and launch n8n
- Importing and configuring a pre-built workflow
- Connecting to Lemonade using the native n8n integration
- Understanding workflow nodes and data flow

## Prerequisites

<!-- @require:lemonade -->

> 

Reviewer 2: As Daniel said, let's add a Lemonade section where we explain:
- what is lemonade? 
- why are we using it instead of LMStudio
- how to set it up

Reviewer 2: mention again that we're using local llm serving to connect to the automation tool

Author: Done

> 

Reviewer 2: node/npm should also be in the prereqs

Reviewer 3: I can add that.

Reviewer 3: Done!

## Installing n8n

> Your STX Halo™ has Node.js pre-installed. Install n8n globally using npm:

Reviewer: ```suggestion
Your STX Halo has Node.js pre-installed. Install n8n globally using npm:
```

```bash
npm install -g n8n
```

## Launching n8n

Start n8n from the terminal:

```bash
n8n start
```

n8n starts a local web server. Open your browser to `http://localhost:5678` to access the editor.

> **Tip**: Keep the terminal window open while using n8n. Closing it will stop the server.

## Setting Up the Workflow

### Step 1: Sign Up or Log In to n8n

When you first open n8n, you'll be prompted to create an account or log in:

1. Open `http://localhost:5678` in your browser
2. Create a new account with your email, or log in if you already have one
3. Once logged in, you'll see the n8n dashboard

### Step 2: Import the Workflow

We've provided a pre-built workflow that you can import directly:

> 1. Download the workflow file: [financial-news-workflow.json](assets/financial-news-workflow.json)

Reviewer 2: is this available anywhere? are assets/ available to the user? 

regardless, it might be good if we explain the json if users will have to edit them in the future.

Author: Yes, this is added to assets. They dont have to edit the json. They will only import the json.

2. In n8n, click **Add workflow** in the top right
3. Click the **...** menu (three dots) and select **Import from file**
4. Select the downloaded `financial-news-workflow.json` file
5. The workflow will appear on the canvas

### Step 3: Understanding the Workflow

The imported workflow contains 7 connected nodes:

<p align="center">
  <img src="assets/workflow-overview.png" alt="n8n Financial News Workflow" width="800"/>
</p>

| Node | Purpose |
|------|---------|
| **When clicking 'Execute workflow'** | Manual trigger to start the workflow |

> | **Fetch Financial News Webpage** | HTTP GET request to `https://www.ft.com/` |

Reviewer 2: just double checking that ft.com is scraper-friendly and that it'll actually work. This will probably get double-checked in pre-AFE by QA and UX.

Author: Yeah I tested it and it works, we use this as a demo pretty often.

| **Delay to Ensure Page Load** | Wait node to ensure page content is fully loaded |
| **Extract News Headlines & Text** | HTML node that extracts headlines, editor's picks, top stories, and regional news using CSS selectors |
| **Clean Extracted News Data** | Set node that combines all extracted data into a single text field |
| **AI Financial News Summarizer** | AI Agent that processes the news with a financial analyst system prompt |

> | **Lemonade Chat Model** | Connects to your local Lemonade server running gpt-oss-20b |

Reviewer 2: lets not say the specific model here since it might change.

Author: Done

| **Structured Output Parser** | Formats the AI output as structured JSON |

> | **Convert to File** | Converts the summary to a downloadable file |

Reviewer 2: this part is good

### Step 4: Configure Lemonade Credentials

Before running the workflow, you need to connect it to your local Lemonade server:

1. Click the **Lemonade Chat Model** node
2. Click **Credential to connect with** → **Create New Credential**
3. Enter the following settings:

| Field | Value |
|-------|-------|
| **Base URL** | `http://localhost:8000/api/v1` |
| **API Key** | `lemonade` |

4. Click **Save**

> > **Note**: Ensure Lemonade is running before testing. The workflow uses `gpt-oss-20b-mxfp4-GGUF` by default—you can change this in the Lemonade Chat Model node settings.

Reviewer 2: make sure this default model remains accurate

### Step 5: Test the Workflow

1. Ensure Lemonade is running with a model loaded
2. Click **Test workflow** in the top right corner
3. Watch each node execute from left to right—they turn green when complete
4. Click the **AI Financial News Summarizer** node to see the generated summary

## Understanding the AI Agent

> 

Reviewer 2: is this editable within the UI? I assume so. Maybe we can mention that to the user so they can start customizing.

Author: All nodes are customizable in its own way. As they start clicking the buttons they will see the fields and at the bottom we talk about exploring other workflows as well.

The AI Financial News Summarizer uses a system prompt designed for financial analysis:

```
You are an AI financial analyst. Your role is to read, understand, and
summarize key financial news from today. The goal is to provide investors
with a clear and concise market overview to support better investment decisions.

Investor Outlook
Today's news points to [bullish/bearish/neutral] sentiment. Watch for
[economic event/earnings report] tomorrow, which could influence market direction.
```

The agent receives the cleaned news data and outputs a structured summary with market sentiment.

## Saving Your Workflow

Click the workflow name at the top and rename it if desired. Workflows auto-save as you work.

## Next Steps

- **Schedule automation**: Replace Manual Trigger with a **Schedule Trigger** to run daily
- **Send notifications**: Add a **Discord**, **Slack**, or **Email** node to receive summaries
- **Try different models**: Change the model in the Lemonade Chat Model node to experiment with different LLMs

> - **Customize extraction**: Modify the HTML Extract node's CSS selectors to target different news sections

Reviewer 2: im thinking we should not lock users into lemonade, and allow them to use LMStudio or Ollama too. We can potentially: - add that at least as a next step: **Try different backends such as LMStudio or Ollama** 
- add resources/documentation that has that
- add a small new section that briefly summarizes the steps to do this, and how to change it within the n8n workflow.

Reviewer 2: also, another next step could be trying different automation platforms. CrewAI and AutoGen might be popular and work too. After all, the main point of this playbook is connecting a backend (lemonade/lmstudio/ollama) with a automation platform (n8n, crewai, autogen) and editing those workflows.

Author: Explaining each step for adding ollama/LMStudio nodes and configuring the credentials is a lot of steps. I will mention using other providers and point to example workflows that use ollama.

n8n includes a **native Lemonade node** (`Lemonade Chat Model`) that provides a first-class integration—no need to configure OpenAI-compatible endpoints manually. This makes connecting your local LLM to automation workflows straightforward.

## Prerequisites

<!-- @require:lemonade,nodejs -->

## Installing n8n

Your STX Halo has Node.js pre-installed. Install n8n globally using npm:

```bash
npm install -g n8n
```

## Launching n8n

Start n8n from the terminal:

```bash
n8n start
```

n8n starts a local web server. Open your browser to `http://localhost:5678` to access the editor.

> **Tip**: Keep the terminal window open while using n8n. Closing it will stop the server.

## Setting Up the Workflow

### Step 1: Sign Up or Log In to n8n

When you first open n8n, you'll be prompted to create an account or log in:

1. Open `http://localhost:5678` in your browser
2. Create a new account with your email, or log in if you already have one
3. Once logged in, you'll see the n8n dashboard

### Step 2: Import the Workflow

We've provided a pre-built workflow that you can import directly:

1. Download the workflow file: [financial-news-workflow.json](assets/financial-news-workflow.json)

> 2. In n8n, click **Add workflow** in the top right

Reviewer: This instruction doesn't match for me -- if it's their first time making an n8n workflow, they have to click "Start from Scratch" first. There's no "Add workflow" button.

Author: Good catch. Fixed.

> 3. Click the **...** menu (three dots) and select **Import from file**

Reviewer: ```suggestion
3. Click the **...** menu (three dots) in the top-right corner and select **Import from file**
```

The imported workflow contains 7 connected nodes:

<p align="center">
  <img src="assets/workflow-overview.png" alt="n8n Financial News Workflow" width="800"/>
</p>

| Node | Purpose |
|------|---------|
| **When clicking 'Execute workflow'** | Manual trigger to start the workflow |
| **Fetch Financial News Webpage** | HTTP GET request to `https://www.ft.com/` |
| **Delay to Ensure Page Load** | Wait node to ensure page content is fully loaded |
| **Extract News Headlines & Text** | HTML node that extracts headlines, editor's picks, top stories, and regional news using CSS selectors |
| **Clean Extracted News Data** | Set node that combines all extracted data into a single text field |
| **AI Financial News Summarizer** | AI Agent that processes the news with a financial analyst system prompt |
| **Lemonade Chat Model** | Connects to your local Lemonade server running the LLM |
| **Structured Output Parser** | Formats the AI output as structured JSON |
| **Convert to File** | Converts the summary to a downloadable file |

### Step 4: Configure Lemonade Credentials

Before running the workflow, you need to connect it to your local Lemonade server:

> 1. Click the **Lemonade Chat Model** node

Reviewer: ```suggestion
1. Double click the **Lemonade Chat Model** node
```

> 2. Click **Credential to connect with** → **Create New Credential**

Reviewer: ```suggestion
2. In the dropdown menu **Credential to connect with** select **Create New Credential**
```

3. Enter the following settings:

| Field | Value |
|-------|-------|
| **Base URL** | `http://localhost:8000/api/v1` |
| **API Key** | `lemonade` |

4. Click **Save**

> > **Note**: Ensure Lemonade is running before testing. The workflow uses `gpt-oss-120b-mxfp4-GGUF` by default—you can change this in the Lemonade Chat Model node settings.

Reviewer: Do we want to add a note on how to download the model if it's not already downloaded? Or are we assuming 1) it's downloaded or 2) user already knows how to do that by this playbook?

Author: The GPT OSS 120B model that we are using is already downloaded to the system.

### Step 5: Test the Workflow

1. Ensure Lemonade is running with a model loaded

> 2. Click **Test workflow** in the top right corner

Reviewer: I didn't see a "Test workflow" just a "Execute workflow" and it was in the center bottom of the page

Author: Fixed
